scripts/find_type.py

```diff
@@ -0,0 +1,67 @@
+# Usage: find_type.py FILENAME START_LINE START_COL END_LINE END_COL MYPY_AND_ARGS
+# Prints out the type of the expression in the given location if the mypy run
+# succeeds cleanly.  Otherwise, prints out the errors encountered.
+# Note: this only works on expressions, and not assignment targets.
+#
+#
+# Example vim usage:
+# function RevealType()
+#   " Set this to the command you use to run mypy on your project.  Include the mypy invocation.
+#   let mypycmd = 'python3 -m mypy mypy --incremental'
+#   let [startline, startcol] = getpos("'<")[1:2]
+#   let [endline, endcol] = getpos("'>")[1:2]
+#   " Change this line to point to the find_type.py script.
+#   execute '!python3 /path/to/mypy/scripts/find_type.py % ' . startline . ' ' . startcol . ' ' . endline . ' ' . endcol . ' ' . mypycmd
+# endfunction
+# vnoremap <Leader>t :call RevealType()<CR>
+
+import subprocess
+import sys
+import tempfile
+
+REVEAL_TYPE_START = 'reveal_type('
+REVEAL_TYPE_END = ')'
+
+def insert(line, s, pos):
+    return line[:pos] + s + line[pos:]
+
+def run_mypy(mypy_and_args, filename, tmp_name):
+    proc = subprocess.run(mypy_and_args + ['--shadow-file', filename, tmp_name], stdout=subprocess.PIPE)
+    return proc.stdout.decode(encoding="utf-8")
+
+def process_output(output, filename, start_line):
+    line_format = "{}:{}: error: Revealed type is '".format(filename, start_line + 1)
+    error_found = False
+    for line in output.splitlines():
+        if line.startswith(line_format):
```

Comment:
Can you make this parsing a bit more sophisticated? Mypy normalizes the filename, so that e.g. if I give it the file `./mypy/stubgen.py`, in the error messages the `./` will be stripped. Ditto for `$PWD/mypy/stubgen.py` and other spellings of the same file. (If the file is not under the current directory it outputs its unchanged IIRC.)

Unless there are a *ton* of errors, comparing using `os.path.samefile()` should be harmless.